rules/hmap.bzl

```diff
@@ -72,20 +72,28 @@ def _make_headermap_impl(ctx):
         namespace = ctx.attr.namespace,
         hdrs_lists = hdrs_lists,
     )
-
+    compilation_context = cc_common.create_compilation_context(
+        headers = depset([headermap]),
+    )
     cc_info_provider = CcInfo(
-        compilation_context = cc_common.create_compilation_context(
-            headers = depset([headermap]),
-        ),
+        compilation_context = compilation_context,
+    )
+    swift_info_provider = swift_common.create_swift_info(
+        modules = [swift_common.create_module(
+            name = ctx.attr.name,
+            clang = swift_common.create_clang_module(
+                compilation_context = compilation_context,
+                module_map = None,
+            ),
+        )],
     )
-
     providers = [
         DefaultInfo(
             files = depset([headermap]),
         ),
         apple_common.new_objc_provider(),
         cc_info_provider,
```

Comment:
Is cc_info_provider still necessary for rules_swift 1.18.0?

Reply:
1.18 def. need it
not sure about 2.1.1, so far it seems that the logic depends on it if it is not a swift file to compile?
based on the commit i think it is `apple_common.new_objc_provider()` we will remove soon as latest bazel don't have Objc provider under apple_common anymore: https://bazel.build/rules/lib/toplevel/apple_common (7.3 still has it)